learning_resources/views.py

```diff
@@ -179,4 +205,104 @@ def get_queryset(self):
         Returns:
             QuerySet of LearningResource objects that are Programs
         """
-        return self._get_base_queryset(resource_type=LearningResourceType.program.value)
+        return self._get_base_queryset(
+            resource_type=LearningResourceType.program.value
+        ).filter(published=True)
+
+
+class LearningPathViewSet(LearningResourceViewSet, viewsets.ModelViewSet):
+    """
+    Viewset for LearningPaths
+    """
+
+    serializer_class = LearningPathResourceSerializer
+    permission_classes = (permissions.HasLearningPathPermissions,)
+
+    def get_queryset(self):
+        """
+        Generate a QuerySet for fetching valid Programs
+
+        Returns:
+            QuerySet of LearningResource objects that are Programs
+        """
+        return self._get_base_queryset(
+            resource_type=LearningResourceType.learning_path.value,
+        )
+
+    def list(self, request, *args, **kwargs):
+        """Get all learningpaths for learningpath editors, public learningpaths for all others"""
+
+        if is_learning_path_editor(request) or is_admin_user(request):
+            queryset = self.get_queryset()
+        else:
+            queryset = self.get_queryset().filter(published=True)
+
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
+    def create(self, request, *args, **kwargs):
+        request.data["readable_id"] = uuid4().hex
+        request.data["resource_type"] = LearningResourceType.learning_path.value
+        return super().create(request, *args, **kwargs)
+
+    def perform_destroy(self, instance):
+        # deindex_staff_list(instance.learning_resource)
+        instance.delete()
+
+
+class ResourceListItemsViewSet(NestedViewSetMixin, viewsets.ModelViewSet):
+    """
+    Viewset for LearningResource related resources
+    """
+
+    permission_classes = (AnonymousAccessReadonlyPermission,)
+    serializer_class = LearningResourceChildSerializer
+    queryset = (
+        LearningResourceRelationship.objects.select_related("child")
+        .prefetch_related(
+            "child__runs",
+            "child__runs__instructors",
+        )
+        .order_by("position")
+    )
+    pagination_class = DefaultPagination
+
+
+class LearningPathItemsViewSet(ResourceListItemsViewSet):
+    """
+    Viewset for LearningPath related resources
+    """
+
+    serializer_class = LearningResourceRelationshipSerializer
+    permission_classes = (permissions.HasLearningPathItemPermissions,)
+
+    def create(self, request, *args, **kwargs):
+        parent_id = kwargs[f"{DRF_NESTED_PARENT_LOOKUP_PREFIX}parent_id"]
```

Comment:
Could you explain this convention a bit?

Reply:
It has to do with the nested serializer functionality described [here](https://chibisov.github.io/drf-extensions/docs/#nested-routes), the LearningResource id  of the parent LearningPath is contained in the request kwargs.  I will move this to a separate function with explanatory docstring.

Reply:
Ok, a couple of thoughts based on that:

- Maybe use [this method instead](https://github.com/chibisov/drf-extensions/blob/f7de5c0278371d35efc3f8ff929fcccdfb641c21/rest_framework_extensions/mixins.py#L69)?
- It probably shouldn't be a configurable setting, since it effects a public API schema and the schema for the frontend.